src/app/api/tasks/tasks.service.ts

```diff
@@ -379,7 +382,25 @@ export class TasksService extends BaseService {
         )
         await notificationService.markAsReadForAllRecipients(updatedTask)
       } else {
-        await notificationService.create(NotificationTaskActions.Completed, updatedTask)
+        // Handle client task completed
+        const internalUsers = await copilot.getInternalUsers()
+        const client = await copilot.getClient(z.string().parse(updatedTask.assigneeId))
+        const relavantInternalUsers = internalUsers.data.filter((iu) => {
+          // Case I: IU has full access
+          if (!iu.isClientAccessLimited) return true
+          // Case II: IU has limited access in which case it must have a valid companyAccessList
+          return iu.companyAccessList?.includes(client.companyId)
+        })
+        const notificationPromises = []
+        const bottleneck = new Bottleneck({ minTime: 250, maxConcurrent: 2 })
+        for (let iu of relavantInternalUsers) {
```

Comment:
Just to understand the context - is there any specific reasons`minTime` and `maxConcurrent` are set to these values?

Reply:
@aatbip 250ms minTime would give us 4 req pools per second, and maxConcurrent gives us 2 req/pool.

So I'm limiting the requests to 8 req/sec. It's not a good idea to max connections out at 20req/sec since if we're carrying out two or more notification requests at a time one would be ratelimited.

8 is a good balance between speed and safety.

Reply:
Makes sense!